de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -501,10 +495,10 @@ public void run() {
 		 * THis method blocks until the runnable got called.
 		 */
 		void waitForExecution() {
-			synchronized (gotExecuted) {
+			synchronized (this) {

```

Comment:
Do we really need synchronization here? Maybe we could just declare gotExecuted volatile and do thread sleep instead of wait and all synchronization stuff?

Reply:
By experience it is generally good to stick with the basic synchronization mechanisms.